pandas-stubs/core/groupby/generic.pyi

```diff
@@ -229,13 +229,13 @@ class DataFrameGroupBy(GroupBy[DataFrame], Generic[ByT]):
     def filter(
         self, func: Callable, dropna: bool = ..., *args, **kwargs
     ) -> DataFrame: ...
+    @overload  # type: ignore[override]
+    def __getitem__(self, key: slice) -> DataFrameGroupBy[ByT]: ...
```

Comment:
do we have tests for `slice` vs `Scalar | tuple[Hashable, ...]` vs `Iterable[Hashable]`?

Reply:
No, and when I tried to write ones for `tuple[Hashable]` and `slice`, I couldn't get them to work at runtime, so I have removed them from the stubs in the next commit.